### PR TITLE
automation: Add help warning about inclusion of default ports

### PR DESCRIPTION
**Pull request by kingthorin**
- CHANGELOG > Added note.
- environment.html > Added warning note.

Fixes zaproxy/zaproxy#7649

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by thc202** [comment]
I thought it was going to be updated the core help (e.g. https://www.zaproxy.org/docs/desktop/ui/dialogs/session/contexts/#include-in-context), while the issue was reported with automation it's not specific to it.

**Comment by kingthorin** [comment]
Oh I guess I can do both :grinning:

**Comment by kingthorin** [comment]
Good to go?

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/environment.html [hunk 1]
```diff
@@ -12,6 +12,9 @@ <H1>Automation Framework - Environment</H1>
 <p>
 The Automation Framework supports all of the <a href="authentication.html">authentication</a> mechanisms supported by ZAP.
 
+<strong>Note</strong> When testing targets that operate on default ports (80 for http, 443 for https), the colon port portion of the URL should not be included.
+Including that portion (for example: http://example.com:80) may result in an inability to crawl or test the target.
```
Comment (thc202) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/environment.html:16]: I think we should say why, the URLs would not be considered to be in context.

**Comment by kingthorin** [comment]
Tweaked.

File: addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/environment.html [hunk 2]
```diff
@@ -11,6 +11,11 @@ <H1>Automation Framework - Environment</H1>
 This section of the YAML configuration file defines the applications which the rest of the jobs can act on.
 <p>
 The Automation Framework supports all of the <a href="authentication.html">authentication</a> mechanisms supported by ZAP.
+<p>
+<strong>Note</strong> When testing targets that operate on default ports (80 for http, 443 for https), the colon port portion of the URL should not be included.
+Including that portion (for example: http://example.com:80) may result in an inability to crawl or test the target. If a 'default' port is specified both 
+browsers and ZAP treat it without the default port being included then it doesn't match the expectation within the Context and there's nothing to interact with 
+as part of the Content.
```
Comment (thc202) [inline addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/environment.html:18]: `Context`?

Comment (kingthorin) [reply addOns/automation/src/main/javahelp/org/zaproxy/addon/automation/resources/help/contents/environment.html:18]: DOH

**Comment by kingthorin** [comment]
Fixed the fix.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**